### ISSUE TITLE
false negative in bugprone-assert-side-effect when method accept a non-const-ref argument.

### ISSUE THREAD

**Issue by HerrCai0907** [llvm/llvm-project#84092]
https://godbolt.org/z/e34x94Enq
```c++
#include <cassert>

struct S {
  bool f(int& i) const;
};

void foo(S s) {
  int a;
  assert(s.f(a));
}
```
